### ISSUE TITLE
NeoVim closes terminal-buffer after running :q

### ISSUE THREAD

**Issue by StephanMeijer** [neovim/neovim#3138]
In NeoVim, I can open a terminal using `:terminal`. But after a while, I want to close the split of the terminal, but not the terminal itself. Terminals in NeoVim act like buffers, but `:q` kills the buffer, *and* the split. I only want to kill the split, not the buffer. `:q` works well with normal (text-)buffers, but not with "terminal buffers". The behaviour is completely different.

**Comment by justinmk** [comment]
Can't reproduce. Maybe you have a misbehaving autocmd? What are steps to reproduce via `nvim -u NONE` ?

NVIM 0.0.0-alpha+201508030024 (compiled Aug  3 2015 12:38:36)
Commit: 9f350e615cba96713666e681c68b002f38efb407

**Comment by StephanMeijer** [comment]
Well, I moved my .nvimrc and my .nvim to another location, so these files won't be loaded. I started NeoVim with `nvim -u NONE`. I opened a file using `:e`. I made a vertical split using `:vs`. In the right side of the split (I used `CTRL + w + l`, I ran `:terminal`. I got out of terminal-mode using `CTRL + \ + n`. I ran `:ls`, output:
```
  1 %a-  "term://.//16777:/usr/bin/zsh" line 47
  2  a   "test.dot"                     line 0
```
I closed the first side of the split, using `:q`, I ran `:ls`, output:
```
  1 %a-  "term://.//16777:/usr/bin/zsh" line 46
  2      "test.dot"                     line 0
```
I thought: `OK, awesome, my .dot-file buffer is not gone!`
My terminal-buffer is still open and visible.
I made a new split, using `:vs`. I ran :bn in the second side of my vertical split. I see the content of file `test.dot`, great.

I navigate to my terminal-buffer, which is open and visible in the first side of the vertical split, using `CTRL + w + h`. I ran `:q`. Now my terminal buffer is invisible. I ran `:ls`:
```
  2 %a   "test.dot"                     line 1
```
My terminal-buffer is gone. I tried to run `:b 1`, NeoVim told me: `E86: Buffer 1 does not exist`.

My terminal-buffer was deleted after running `:q` while I was in a split, but if I run the same command while in a split on a text-file-buffer, the text-file-buffer won't be deleted. So: Terminal-buffers are acting different, compared to text-file-buffers.
